### PR TITLE
fix: vendor go-yaml

### LINKED ISSUE #2039: run failed: could not unmarshal lockfile: : yaml: line 9047: did not find expected ',' or '}'
### What version of Turborepo are you using?

1.5.1

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Mac

### Describe the Bug

image

image

The most recent work version is 1.4.6

### Expected Behavior

No error

### To Reproduce

Private project, I will create a reproduce repository when I have time

### PR DESCRIPTION
**Pull request by chris-olszewski**
Fixes #2039 

Until I'm able to get [this fix](https://github.com/go-yaml/yaml/pull/905) merged into `go-yaml`, I'm vendoring the library so we don't trip over some of the pnpm lockfiles.

### CODE REVIEW

File: cli/internal/yaml/scannerc.go [hunk 1]
```diff
@@ -0,0 +1,3040 @@
...
+		}
+
+		// Consume non-blank characters.
+		for !is_blankz(parser.buffer, parser.buffer_pos) {
+
+			// Check for indicators that may end a plain scalar.
+			if (parser.buffer[parser.buffer_pos] == ':' && is_blankz(parser.buffer, parser.buffer_pos+1)) ||
+				(parser.flow_level > 0 &&
+					(parser.buffer[parser.buffer_pos] == ',' ||
+						parser.buffer[parser.buffer_pos] == '[' ||
```
Comment (NicholasLYang) [inline cli/internal/yaml/scannerc.go:2731]: Okay this is the *tiniest* little nit, but according to the spec in 7.3.3 it seems like we cannot allow `?` as the first character without it following a "non-space 'safe' character":

> Plain scalars must not begin with most [indicators](https://yaml.org/spec/1.2.2/#indicator-characters), as this would cause ambiguity with other YAML constructs. However, the “:”, “?” and “-” [indicators](https://yaml.org/spec/1.2.2/#indicator-characters) may be used as the first character if followed by a non-[space](https://yaml.org/spec/1.2.2/#white-space-characters) “safe” character, as this causes no ambiguity.

If I'm reading this code correctly, we don't ensure this. It might be worth adding a check and a test too.

File: cli/.golangci.yml [hunk 2]
```diff
@@ -5,6 +5,8 @@ run:
   timeout: 5m
   modules-download-mode: readonly
   go: 1.17
+  skip-dirs:
```
Comment (nathanhammond) [inline cli/.golangci.yml:8]: TIL, love it.

File: cli/internal/yaml/scannerc.go (hunk 1 above)
Comment (nathanhammond) [reply cli/internal/yaml/scannerc.go:2731]: @NicholasLYang is _technically_ correct, the best kind of correct. (He has a small issue in his serialization of the issue about precise order, but the note is valid.) Concretely, all of these tests should pass. The `?`-initials don't pass.

The `? ` (question followed by space) things _do_ do the right thing upstream and are not something to address here.

```go
// Question mark in plain scalar flow collection
{
  "seq: [a?b]",
  map[string]interface{}{"seq": []interface{}{"a?b"}},
}, {
  "seq: [?ab]",
  map[string]interface{}{"seq": []interface{}{"a?b"}},
}, {
  "map: {a: b?c}",
  map[string]interface{}{"map": map[string]interface{}{"a": "b?c"}},
}, {
  "map: {a: ?bc}",
  map[string]interface{}{"map": map[string]interface{}{"a": "?bc"}},
},
```

However, as the change you've made is strictly an improvement (covering two of these four cases, and not regressing anything else), I don't find it required to account for every possible edge case.

**Review by nathanhammond — APPROVED** [review]
As the change you've made is strictly an improvement (covering two of the four cases, and not regressing anything else), I don't find it required to account for every possible edge case.

So: this is a ship, but if you see an additional one-line fix for the remaining two test-cases we'd obviously like that too.

**Review by NicholasLYang — APPROVED** [review]
